docs/user/airbrakes.rst

```diff
@@ -0,0 +1,431 @@
+   - ``y``: The y position of the rocket, in meters.
+   - ``z``: The z position of the rocket, in meters.
+   - ``v_x``: The x component of the velocity of the rocket, in meters per 
+     second.
+   - ``v_y``: The y component of the velocity of the rocket, in meters per 
+     second.
+   - ``v_z``: The z component of the velocity of the rocket, in meters per 
+     second.
+   - ``e0``: The first component of the quaternion representing the rotation 
+     of the rocket.
+   - ``e1``: The second component of the quaternion representing the rotation 
+     of the rocket.
+   - ``e2``: The third component of the quaternion representing the rotation 
+     of the rocket.
+   - ``e3``: The fourth component of the quaternion representing the rotation 
+     of the rocket.
+   - ``w_x``: The x component of the angular velocity of the rocket, in 
+     radians per second.
+   - ``w_y``: The y component of the angular velocity of the rocket, in 
+     radians per second.
+   - ``w_z``: The z component of the angular velocity of the rocket, in 
+     radians per second.
+
+4. ``state_history`` (list): A record of the rocket's state at each
+   step throughout the simulation. The state_history is organized as
+   a list of lists, with each sublist containing a state vector. The
+   last item in the list always corresponds to the previous state
+   vector, providing a chronological sequence of the rocket's
+   evolving states.
+5. ``observed_variables`` (list): A list containing the variables that
+   the controller function returns. The return of each controller
+   function call is appended to the observed_variables list. The
+   initial value in the first step of the simulation of this list is
+   provided by the ``initial_observed_variables`` argument.
+6. ``air_brakes`` (AirBrakes): The ``AirBrakes`` instance being controlled.
+    
+Our example ``controller_function`` will deploy the air brakes when the rocket
+reaches 1500 meters above the ground. The deployment level will be function of the
+vertical velocity at the current time step and of the vertical velocity at the
+previous time step.
+
+Also, the controller function will check for the burnout of the rocket's motor 
+and only deploy the air brakes if the rocket has reached burnout. 
+
+Then, a limitation for the opening/closing speed of the air brakes will be set.
+The air brakes deployment level will not be able to change faster than 20% per
+second, in our case.
+
+Lets define the controller function:
+
+.. jupyter-execute::
+
+    def controller_function(
+        time, sampling_rate, state, state_history, observed_variables, air_brakes
+    ):
+        # state = [x, y, z, vx, vy, vz, e0, e1, e2, e3, wx, wy, wz]
+        altitude_ASL = state[2]
+        altitude_AGL = altitude_ASL - env.elevation
+        vx, vy, vz = state[3], state[4], state[5]
+        mach_number = (vx**2 + vy**2 + vz**2) ** 0.5 / env.speed_of_sound(
+            altitude_AGL - env.elevation
+        )
```

Comment:
mmmm the mach number should be calculated considering the freestream speed instead of the rocket velocities, right?

In a more sophisticated fashion, we would do `mach = ((vx - wind_u)**2 + (vy - wind_v)**2 + ...)**0.5`.

Te current function works when the wind is zero only, which is the case of the simulation in this page.

Reply:
Oof, you are right. I'll add the right calculation for the sake of correctness, but this is a problem with the current controller implementation, and maybe actually the flight class.

Currently our state vector has `(vx vy vz)`, and I believe it would be better to instead use `(free_stream_speed attack_angle sideslip_angle)`. This seems to be the standard when dealing with aerodynamics (aviation uses this as well)

Reply:
The state vector is constraining a lot of our developments in the Flight class lately.
I'm sure soon enough we will be modifying it somehow.